src/utils/metrics.rs

```diff
@@ -0,0 +1,167 @@
+use cadence::{BufferedUdpMetricSink, Counted, Gauged, QueuingMetricSink, StatsdClient, Timed};
+use lazy_static::lazy_static;
+use parking_lot::RwLock;
+use std::collections::HashMap;
+use std::net::{ToSocketAddrs, UdpSocket};
+use std::sync::Arc;
+
+pub trait Metrics {
+    fn counter(&self, key: &str, value: Option<i64>, tags: Option<HashMap<&str, &str>>);
+
+    fn gauge(&self, key: &str, value: u64, tags: Option<HashMap<&str, &str>>);
+
+    fn time(&self, key: &str, value: u64, tags: Option<HashMap<&str, &str>>);
+}
+
+pub struct MetricsClient {
+    pub statsd_client: StatsdClient,
+    prefix: String,
+}
+
+impl Metrics for MetricsClient {
+    fn counter(&self, key: &str, value: Option<i64>, tags: Option<HashMap<&str, &str>>) {
+        let mut count_value: i64 = 1;
+        if let Some(value) = value {
+            count_value = value;
+        }
+
+        if let Some(tags) = tags {
+            let mut metric_builder = self.statsd_client.count_with_tags(key, count_value);
+            for (key, value) in tags {
+                metric_builder = metric_builder.with_tag(key, value);
+            }
+            let result = metric_builder.try_send();
+            match result {
+                Ok(_) => {}
+                Err(_err) => {
+                    println!("Failed to send metric {}: {}", key, _err)
+                }
+            }
+        } else {
+            let result = self.statsd_client.count(key, count_value);
+            match result {
+                Ok(_) => {}
+                Err(_err) => {
+                    println!("Failed to send metric {}: {}", key, _err)
+                }
+            }
+        }
+    }
+
+    fn gauge(&self, key: &str, value: u64, tags: Option<HashMap<&str, &str>>) {
+        if let Some(tags) = tags {
+            let mut metric_builder = self.statsd_client.gauge_with_tags(key, value);
+            for (key, value) in tags {
+                metric_builder = metric_builder.with_tag(key, value);
+            }
+            let result = metric_builder.try_send();
+            match result {
+                Ok(_) => {}
+                Err(_err) => {
+                    println!("Failed to send metric {}: {}", key, _err)
+                }
+            }
```

Comment:
It seems this is the same in all types of metrics. Why not having a function for it ?
```
fn send_with_tags(builder: WhateverTheTypeOfMetricsBuilderIs, tags:Hashmap...)
```